test/docker-e2e/docker_test.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/celestiaorg/tastora/framework/testutil/toml"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap/zaptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	// testChainID is the chain ID used for testing.
+	// it must be the string "test" as it is handled explicitly in app/node.
+	testChainID = "test"
+)
+
+func init() {
+	sdkConf := sdk.GetConfig()
+	sdkConf.SetBech32PrefixForAccount("celestia", "celestiapub")
+	sdkConf.Seal()
+}
+
+func TestDockerSuite(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping due to short mode")
+	}
+	suite.Run(t, new(DockerTestSuite))
+}
+
+type DockerTestSuite struct {
+	suite.Suite
+	provider     tastoratypes.Provider
+	celestia     tastoratypes.Chain
+	daNetwork    tastoratypes.DataAvailabilityNetwork
+	rollkitChain tastoratypes.RollkitChain
+}
+
+// ConfigOption is a function type for modifying tastoradocker.Config
+type ConfigOption func(*tastoradocker.Config)
+
+// CreateDockerProvider creates a new tastoratypes.Provider with optional configuration modifications
+func (s *DockerTestSuite) CreateDockerProvider(opts ...ConfigOption) tastoratypes.Provider {
+	t := s.T()
+	encConfig := testutil.MakeTestEncodingConfig(auth.AppModuleBasic{}, bank.AppModuleBasic{})
+	numValidators := 1
+	numFullNodes := 0
+	client, network := tastoradocker.DockerSetup(t)
+
+	cfg := tastoradocker.Config{
+		Logger:          zaptest.NewLogger(t),
+		DockerClient:    client,
+		DockerNetworkID: network,
+		ChainConfig: &tastoradocker.ChainConfig{
+			ConfigFileOverrides: map[string]any{
+				"config/app.toml":    appOverrides(),
+				"config/config.toml": configOverrides(),
+			},
+			Type:          "celestia",
+			Name:          "celestia",
+			Version:       "v4.0.0-rc6",
+			NumValidators: &numValidators,
+			NumFullNodes:  &numFullNodes,
+			ChainID:       testChainID,
+			Images: []tastoradocker.DockerImage{
+				{
+					Repository: "ghcr.io/celestiaorg/celestia-app",
+					Version:    "v4.0.0-rc6",
+					UIDGID:     "10001:10001",
+				},
+			},
+			Bin:            "celestia-appd",
+			Bech32Prefix:   "celestia",
+			Denom:          "utia",
+			CoinType:       "118",
+			GasPrices:      "0.025utia",
+			GasAdjustment:  1.3,
+			EncodingConfig: &encConfig,
+			AdditionalStartArgs: []string{
+				"--force-no-bbr",
+				"--grpc.enable",
+				"--grpc.address",
+				"0.0.0.0:9090",
+				"--rpc.grpc_laddr=tcp://0.0.0.0:9098",
+				"--timeout-commit", "1s",
+			},
+		},
+		DataAvailabilityNetworkConfig: &tastoradocker.DataAvailabilityNetworkConfig{
+			BridgeNodeCount: 1,
+			Image: tastoradocker.DockerImage{
+				Repository: "ghcr.io/celestiaorg/celestia-node",
+				Version:    "pr-4283",
```

Comment:
has a fix which enables the signer type access wallets funded after initialization